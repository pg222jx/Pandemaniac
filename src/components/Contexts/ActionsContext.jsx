import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

