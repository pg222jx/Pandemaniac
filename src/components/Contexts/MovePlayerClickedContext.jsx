import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

