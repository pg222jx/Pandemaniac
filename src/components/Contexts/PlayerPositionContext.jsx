import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

