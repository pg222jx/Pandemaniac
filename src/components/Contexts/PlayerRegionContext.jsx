import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

