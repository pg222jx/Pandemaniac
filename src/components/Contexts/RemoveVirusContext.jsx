import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

