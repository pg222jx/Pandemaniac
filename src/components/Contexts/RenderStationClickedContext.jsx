import React,{ useState, createContext } from 'react';

export const RenderStationClickedContext = createContext()

export const RenderStationClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RenderStationClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RenderStationClickedContext.Provider>
    );
}

