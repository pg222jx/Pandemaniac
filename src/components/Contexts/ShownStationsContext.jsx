import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

