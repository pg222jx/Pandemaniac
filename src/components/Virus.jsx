/**
 * Representing the virus.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useContext } from 'react'
import virusImage from '../images/virus.png'
import { RemoveVirusContext } from './Contexts/RemoveVirusContext'
import { ActionsContext } from './Contexts/ActionsContext'
import { PlayerRegionContext } from './Contexts/PlayerRegionContext' 

const Virus = (props) => {
    
    const [removeVirusClicked, setRemoveVirusClicked] = useContext(RemoveVirusContext)
    const [amount, setAmount] = useContext(ActionsContext)
    const [region, setRegion] = useContext(PlayerRegionContext)// eslint-disable-line no-unused-vars

    const style = {bottom: props.bottom, left: props.left, visibility: props.visibility}
    const virusId = props.virusid
    const virusRegion = virusId.slice(0, 2)
    
    /**
     * Checks so the player is in same position as the virus it wants to remove.
     */
    const checkPlayerPosition = () => {
        if (virusRegion === region) {
            checkIfClicked()
        }
    }
    /**
     * Removes virus if the "remove virus" button is clicked and actions is not 0.
     */
    const checkIfClicked = () => {
        if(amount === 0) {
            props.information.textContent = 'No more actions. Click "Done with turn".'
            return
        }
        if (removeVirusClicked === true && amount > 0 ) {
            props.remove(virusId)
            setRemoveVirusClicked(false)
            setAmount(amount -1)
        }
    }

    return (
        <img src={virusImage} alt="Virus" className="virus" style={style} virusid={virusId} onClick={() => {checkPlayerPosition()}}/>
    )
}

export default Virus
