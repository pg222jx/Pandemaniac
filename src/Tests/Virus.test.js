import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { RemoveVirusProvider } from '../components/Contexts/RemoveVirusContext'
import { ActionsProvider } from '../components/Contexts/ActionsContext'
import { PlayerRegionProvider } from '../components/Contexts/PlayerRegionContext'
import Virus from '../components/Virus'


const wrapperShallow = shallow(
    <RemoveVirusProvider>
        <ActionsProvider>
            <PlayerRegionProvider>
                <Virus virusid={'NA 1'}/>
            </PlayerRegionProvider>
        </ActionsProvider>
    </RemoveVirusProvider>
)


describe('testing the Virus component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <RemoveVirusProvider>
                <ActionsProvider>
                    <PlayerRegionProvider>
                        <Virus virusid={'NA 1'}/>
                    </PlayerRegionProvider>
                </ActionsProvider>
            </RemoveVirusProvider>, div
        )
       })
})
