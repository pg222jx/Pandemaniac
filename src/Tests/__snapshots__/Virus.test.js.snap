// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Virus component should match the snapshot 1`] = `"<img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" virusid=\\"NA 1\\"/>"`;
